### PR TITLE
Drop execution of eventing-kafka tests

### PR DESCRIPTION
**Pull request by mgencur**
* Removes the repo from the source image
* Removes bash functions for executing upstream tests

Note: Upgrade tests and serverless-operator E2E tests still depend on eventing-kafka through vendoring.

Fixes https://issues.redhat.com/browse/SRVKE-1382

## Proposed Changes

-
-
-

### CODE REVIEW

File: hack/lib/vars.bash [hunk 1]
```diff
@@ -20,7 +20,6 @@ export KNATIVE_EVENTING_VERSION="${KNATIVE_EVENTING_VERSION:-$(metadata.get depe
 export KNATIVE_EVENTING_VERSION_PREVIOUS="${KNATIVE_EVENTING_VERSION_PREVIOUS:-$(metadata.get dependencies.previous.eventing)}"
 # TODO(matzew): SRVKE-1076 remove this when kafka e2e tests have been migrated
 export KNATIVE_EVENTING_KAFKA_VERSION="${KNATIVE_EVENTING_KAFKA_VERSION:-v$(metadata.get dependencies.eventing_kafka)}"
```
[the remark this answers is not in the document]
Comment (matzew) [reply hack/lib/vars.bash:22]: We seem to not use it on "EKB" repo. Perhaps we should do something about moving this? Or using else?

[the remark this answers is not in the document]
Comment (mgencur) [reply hack/lib/vars.bash:22]: AFAIK, "EKB" is independent of this old implementation. We still support it but it doesn't mean we should move it's code and very specific tests into EKB.
EKB itself also depends on eventing-kafka through vendoring. It's same for serverless-operator - it depends on specific parts of eventing-kafka (like this specific test.
I'd probably keep it as it is but not totally sure.
What did you mean by "Or using else"?

**Comment by mgencur** [comment]
/test 4.12-upgrade-tests-aws-ocp-412
Unrelated failure.

File: hack/lib/vars.bash (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgencur) [reply hack/lib/vars.bash]: Pushed a new commit to remove the wathola-kafka-sender:
It was only used with old Kafka Source continual tests but they were
replaced a long time ago with those from EKB.

[the remark this answers is not in the document]
Comment (mgencur) [reply hack/lib/vars.bash]: We still execute some "continual" upgrade tests for the old implementation and the tests are coming from eventing-kafka (through vendoring). But we don't need any extra image for that.

**Comment by mgencur** [comment]
@matzew Rebased. WDYT about current state?

File: olm-catalog/serverless-operator/project.yaml [hunk 2]
```diff
@@ -47,7 +47,6 @@ dependencies:
   # eventing core midstream branch name
   eventing_artifacts_branch: release-v1.8
 
-  eventing_kafka: 1.1.0
```
Comment (matzew) [inline olm-catalog/serverless-operator/project.yaml:50]: sweet!

**Review by matzew — APPROVED** [review]
This looks super!

thanks a lot, @mgencur 

/lgtm
/approve

**Comment by mgencur** [comment]
Ouch, merge conflict. Let me rebase.

**Review by matzew — APPROVED** [review]
/lgtm
/approve

/retest

**Pull request merged**